minetest/minetest
# WIP: Add formspec rendering tests

The aim of this PR is to reduce the likelihood of formspec rendering regressions by implementing automated unit tests. This requires #9188 to be merged first


## To do

This PR is a Work in Progress

- [ ] Add tolerance to image comparisons
- [x] Make commit only change different images

## How to test

* Make a new test by adding a formspec file (.fs) in `util/fstests/tests`
* Run the tests `./util/fstest/run.sh`
* See the result file which was generated for your new test at `util/fstests/reports`
* Commit the result file as an expected output by using `./util/fstest/apply.sh`
